src/graphql_qb/dataset_model.clj

```diff
@@ -0,0 +1,182 @@
+      "PREFIX rdfs: <http://www.w3.org/2000/01/rdf-schema#>"
+      "PREFIX qb: <http://purl.org/linked-data/cube#>"
+      "SELECT ?dim ?label ?comment WHERE {"
+      "  ?dim a qb:DimensionProperty ."
+      "  ?dim <" (str dataset-label) "> ?label ."
+      "}")))
+
+(defn dimension-bindings->dimensions [dimension-bindings configuration]
+  (map (fn [[dim-uri bindings]]
+         (let [labels (map :label bindings)]
+           {:uri dim-uri
+            :label (util/find-best-language labels (config/schema-label-language configuration))}))
+       (group-by :dim dimension-bindings)))
+
+(defn find-all-dimensions [repo configuration]
+  (let [q (find-all-dimensions-query configuration)
+        results (util/eager-query repo q)]
+    (dimension-bindings->dimensions results configuration)))
+
+(defn measure-bindings->measures [measure-bindings configuration]
+  (map (fn [[measure-uri bindings]]
+         (let [labels (map :label bindings)]
+           {:uri measure-uri
+            :label (util/find-best-language labels (config/schema-label-language configuration))}))
+       (group-by :measure measure-bindings)))
+
+(defn find-all-measures-query [configuration]
+  (let [dataset-label (config/dataset-label configuration)]
+    (str
+      "PREFIX rdfs: <http://www.w3.org/2000/01/rdf-schema#>"
+      "PREFIX qb: <http://purl.org/linked-data/cube#>"
+      "SELECT ?measure ?label WHERE {"
+      "  ?measure a qb:MeasureProperty ."
+      "  ?measure <" (str dataset-label) "> ?label ."
+      "}")))
+
+(defn- is-measure-numeric? [repo measure-uri]
+  (sp/query "is-measure-numeric.sparql" {:measure measure-uri} repo))
+
+(defn find-numeric-measures [repo all-measures]
+  (into #{} (keep (fn [{:keys [uri] :as measure}]
+                    (if (is-measure-numeric? repo uri)
+                      uri))
+                  all-measures)))
+
+(defn find-all-measures [repo configuration]
+  (let [q (find-all-measures-query configuration)
+        results (util/eager-query repo q)
+        measures (measure-bindings->measures results configuration)
+        numeric-measures (find-numeric-measures repo measures)]
+    (map (fn [{:keys [uri] :as measure}]
+           (assoc measure :is-numeric? (contains? numeric-measures uri)))
+         measures)))
+
+(defn- get-all-components [repo]
+  (util/distinct-by :comp (sp/query "get-all-components.sparql" repo)))
+
+(defn get-codelist-members-query [configuration]
+  (let [dimension-filters (map (fn [dim] (str "FILTER(?dim != <" dim ">)")) (config/ignored-codelist-dimensions configuration))]
+    (str
+      "PREFIX qb: <http://purl.org/linked-data/cube#>"
+      "PREFIX skos: <http://www.w3.org/2004/02/skos/core#>"
+      "SELECT * WHERE {"
+      "  ?ds a qb:DataSet ."
+      "  ?ds qb:structure ?dsd ."
+      "  ?dsd a qb:DataStructureDefinition ."
+      "  ?dsd qb:component ?comp ."
+      "  ?comp qb:dimension ?dim ."
+      (string/join "\n" dimension-filters)
+      "  ?comp qb:codeList ?codelist ."
```

Comment:
The qb:codeList can be aither at the ?comp or the ?dim. We should use the `:codelist-source` from the configuration